picard/ui/profileeditor.py

```diff
@@ -0,0 +1,455 @@
+        self.ui.buttonbox.rejected.connect(self.close)
+
+        self.ui.buttonbox.addButton(QtWidgets.QDialogButtonBox.Help)
+        self.ui.buttonbox.helpRequested.connect(self.show_help)
+
+    def load(self):
+        """Load initial configuration.
+        """
+        config = get_config()
+        self.profile_settings = config.profiles[self.SETTINGS_KEY]
+
+        for profile in config.profiles[self.PROFILES_KEY]:
+            list_item = ProfileListWidgetItem(profile['title'], profile['enabled'], profile['id'])
+            self.ui.profile_list.addItem(list_item)
+
+        # Select the last selected profile item
+        last_selected_profile_pos = config.persist[self.POSITION_KEY]
+        self.expanded_sections = config.persist[self.EXPANDED_KEY]
+        last_selected_profile = self.ui.profile_list.item(last_selected_profile_pos)
+        settings = None
+        if last_selected_profile:
+            self.ui.profile_list.setCurrentItem(last_selected_profile)
+            last_selected_profile.setSelected(True)
+            id = last_selected_profile.profile_id
+            self.current_profile_id = id
+            settings = self.get_settings_for_profile(id)
+        self.make_setting_tree(settings=settings)
+
+    def get_settings_for_profile(self, id):
+        """Get the settings for the specified profile ID.
+
+        Args:
+            id (str): ID of the profile
+
+        Returns:
+            dict: Profile settings
+        """
+        # Add empty settings dictionary if no dictionary found for the profile.
+        if id not in self.profile_settings:
+            self.profile_settings[id] = {}
+        return self.profile_settings[id]
+
+    def get_current_selected_item(self):
+        """Gets the profile item currently selected in the profiles list.
+
+        Returns:
+            ProfileListWidgetItem: Currently selected item
+        """
+        items = self.ui.profile_list.selectedItems()
+        if items:
+            return items[0]
+        return None
+
+    def update_current_expanded_items_list(self):
+        if self.building_tree:
+            return
+        self.expanded_sections = []
+        for i in range(self.ui.settings_tree.topLevelItemCount()):
+            tl_item = self.ui.settings_tree.topLevelItem(i)
+            if tl_item.isExpanded():
+                self.expanded_sections.append(tl_item.text(0))
+
+    def profile_selected(self, update_settings=True):
+        """Update working profile information for the selected item in the profiles list.
+
+        Args:
+            update_settings (bool, optional): Update settings tree. Defaults to True.
+        """
+        item = self.get_current_selected_item()
+        if item:
+            id = item.profile_id
+            self.current_profile_id = id
+            settings = self.get_settings_for_profile(id)
+            if update_settings:
+                self.make_setting_tree(settings=settings)
```

Comment:
Since `settings` is only used in this `if` branch, its declaration should be moved here.